lib/grandpa/message.go

```diff
@@ -77,21 +77,57 @@ func (v *VoteMessage) ToConsensusMessage() (*ConsensusMessage, error) {
 	}, nil
 }
 
-// NeighbourMessage represents a network-level neighbour message
-type NeighbourMessage struct {
-	Version byte
-	Round   uint64
-	SetID   uint64
-	Number  uint32
+// VersionedNeighbourPacket represents the enum of nighbor messages
+type VersionedNeighbourPacket scale.VaryingDataType
+
+// Index Returns VDT index
+func (VersionedNeighbourPacket) Index() uint { return 2 }
+
+func newVersionedNighborMessage() VersionedNeighbourPacket {
+	vdt := scale.MustNewVaryingDataType(NeighbourPacketV1{})
+
+	return VersionedNeighbourPacket(vdt)
+}
+
+// Set updates the current VDT value to be `val`
+func (vnm *VersionedNeighbourPacket) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*vnm)
+	err = vdt.Set(val)
+	if err != nil {
+		return
+	}
+	*vnm = VersionedNeighbourPacket(vdt)
+	return
```

Comment:
nit naked returns just makes things confusing imo

```suggestion
	return nil
```